pkg/target/cluster.go

```diff
@@ -80,6 +101,55 @@ func (cl *Cluster) Send(cwg *sync.WaitGroup, finish chan struct{}) {
 	}()
 }
 
+func (cl *Cluster) removeAvailableHost(host *Host) {
+	for i, h := range cl.AvailableHosts {
+		if h.Name == host.Name {
+			host.stateChanges.Inc()
+			cl.Hm.Lock()
+			defer cl.Hm.Unlock()
+			length := len(cl.AvailableHosts)
+
+			for j := i; j < length-2; j++ {
+				cl.AvailableHosts[i] = cl.AvailableHosts[i+1]
```

Comment:
```suggestion
				cl.AvailableHosts[j] = cl.AvailableHosts[j+1]
```

Reply:
How about adding tests for manipulation of AvailableHosts?

Reply:
Added tests, which now shuffle order of hosts and call remove on them. Helped fix a bug.